### PR TITLE
[Ready] Fix virus immunity to work and minor diseases tweaks.

### LINKED ISSUE #895: VIRUSES AREN'T CURABLE AT ALL
### Expected Behaviour

Cure a virus and develop antibodies so you can no longer get a disease
### Actual Behaviour

Everyone on the station has brain AIDS or fucking skin burn that doesn't go away ever
### Steps to reproduce the problem
1. Get level 7 biohazard virus
2. Cure it
3. It comes back
4. ask admin to cure it
5. IT DOESN'T GO AWAY

If this is a dupe of another issue then fuck me
Critical priority also pls

### PR DESCRIPTION
**Pull request by Nirnael**
### Fix #895 #716, adds onto #951. DONE VERY MINOR ADDITIONS, JUST FIX

No bullshittery, just fixes those, with some minor additions such as increasing cure chance of transformation diseases and removing ability to cure when transformed.

:cl:
bugfix: Fixes diseases not being curable.
tweak: Transformation diseases can no longer be cured after stage 5, they stay on the mob until it's deleted.
tweak: Zombie and jungle fever cure chance increased.
tweak: You can no longer instantly heal by ingesting cure, it now properly takes into account the cure chance of the disease.
/:cl:

> This goes for the static diseases, not the ones made in viro, I'll look into those separately.
> Diseases are now thoroughly fixed:
> - Curing them actually takes time as opposed to being instant upon ingesting the cure.
> - All diseases can be cured and resisted upon cure, except transformation ones(borg, corgi, xeno, zombie, monkey, etc.) at final stage, once you get transformed there's no going back (except borging, genetics, the usual stuff). This buffs them so that they aren't a mild inconvenience, but actually require them to be attended to.
> - Being fed no longer helps against spread and spaceacillin now only lowers the spread chance to a third instead of outright removing it.
> 
> **What I'm planning on doing with diseases:**
> - Make cures lower the stage chance advancement, and spaceacillin, but by a little bit. 
> - Make vitamins lower the stage chance and react to viro custom-made diseases by removing symptoms, opposing them, etc.
> - Make diseases more resistant to cure as stages advance.
> - Make cryo stop disease advancement (not sure if it's already in the game actually) and diseases respond to temperature.
> - Add some of these to viro custom diseases.
> - Punish the crew for not having the janitor clean the bodily fluids.
> - Add more diseases.
> - Make rage virus not be zombie, make zombie its own gamemode (waiting to see if Kn0ss0s is on it), instead make it something like in the film Kingsman where people just kill each other (obviously, needs RP and rules and reducing murderbone...possibly).
> - Ideally make airborne viruses spread through atmos and add evolution to viro diseases.
> 
> In any case I'll be looking into the first three tomorrow and friday, so tell me what you think about the rest and any other input you have. My general view is to make the crew suffer if they ignore diseases, and not just meta/powergame the cure in 1 minute and continue fucking around.

### CODE REVIEW

**Comment by oisin100** [comment]
> Punish the crew for not having the janitor clean the bodily fluids.

Blood already transmits blood diseases. This used to be such a pain with zombies...

> Make rage virus not be zombie, make zombie its own gamemode (waiting to see if Kn0ss0s is on it), instead make it something like in the film Kingsman where people just kill each other (obviously, needs RP and rules and reducing murderbone...possibly).

It would be nice to see the zombies gamemode behave a bit like when we ran the first zombie events. All the lights go out. One king zombie. Crew holding out against zombies in the brig.

**Comment by Nirnael** [comment]
I meant something like making blood and vomit slippery, and tweaking the infection cause it's static range atm. And if you slip in those you carry it on your clothes to other people.

**Comment by ghost** [comment]
> slippery blood

good luck chasing the guy eswording the captain down in maint because you keep fucking tripping on the captain's anal prolapse aftermath

**Comment by Nirnael** [comment]
Truly needs balancing, maybe various overlays will slip only, but vomit definitely would, or at least shoes would carry it if you step in them.

**Comment by oisin100** [comment]
> Truly needs balancing, maybe various overlays will slip only, but vomit definitely would, or at least shoes would carry it if you step in them.

This PR was meant for fixing the diseases. Perhaps open an issue asking whether it should be a thing.

 I could only ever see this being balanced if it only slipped people not wearing shoes, and then i would just feel sorry for the people who get cloned and have to try and get out of genetics without being slipped multiple times.

**Comment by Nirnael** [comment]
Yeah I chose to not do it, i just modified cure stage and infection chance and i'll leave it at that for now. I'm running tests atm so it should be ready soon.

File: code/datums/diseases/_disease.dm [hunk 1]
```diff
@@ -104,7 +103,7 @@ var/list/diseases = typesof(/datum/disease) - /datum/disease
 	if(affected_mob)
 		if( affected_mob.reagents.has_reagent("spaceacillin") || (affected_mob.satiety > 0 && prob(affected_mob.satiety/10)) )
 			return
-
+
```
Comment (george99g) [inline code/datums/diseases/_disease.dm:106]: Useless random space?

Comment (Nirnael) [inline code/datums/diseases/_disease.dm:106]: For some reason git does that shit, I have no idea why.

**Pull request merged**